### DISCUSSION TITLE
Add react-select inside a custom node

### DISCUSSION THREAD

**Discussion by dariolongo-dufercodev** [xyflow/xyflow#2694] [Q&A] [answered]
Hello,
first of all, thanks for the great works of react-flow!

I'm trying to add a select component (react-select) in a custom node:
[https://codesandbox.io/s/kind-firefly-j80jl6?file=/src/CustomNode.js:573-601](https://codesandbox.io/s/kind-firefly-j80jl6?file=/src/CustomNode.js:573-601)

The select doesn't open on click.

If I force the props "menuIsOpen" of react-select, the dropdown is visible and is possible choose the option with the keyboard.
If I try with with a classic input text the click works fine and set the focus. (I've try also with a datepicker)

I've tried with:
- preventDefault and stopPropagation in onNodeClick event
- draggable/selectable false in the custom node
but nothing help...

If anyone can help me, I'd be grateful.
Thanks!

PS:
"reactflow": "^11.2.0",
"react": "^17.0.2",

**Comment by moklick** [comment, marked as answer]
Hey! Just add the class name "nodrag" so that React Flow doesn't swallow the click event.

**Reply by dariolongo-dufercodev** [reply]
Hi,
thanks for the replay, it works!
Just a note: I've to set ```draggable: true``` for the new node, and the class name "nodrag". If the new node config contain ```draggable: false```, the dropdown doesn't open.

**Reply by njorqueraAssayware** [reply]
Is there any other way to do this? I just want a select inside the node, but its important the node can be draggable and selectable. Im thinking in a way to make the node not selectable and draggable just in the select part, in that way it should work fine

**Reply by njorqueraAssayware** [reply]
in case someone is wondering, i did it. i tried what i was thinking but it didnt work, so instead i just add the className = 'nodrag' when click the select

**Reply by maxwidergy** [reply]
If i add the nodrag classname to the element, then the select swallows the click and reactflow doesnt set the selected property of my node. Anyone could find a way to fix this?

**Reply by moklick** [reply]
@dariolongo-dufercodev when the node is set to `draggable: false` you need to add the class name "nopan" instead of "nodrag".

**Reply by jaf7** [reply]
> @dariolongo-dufercodev when the node is set to `draggable: false` you need to add the class name "nopan" instead of "nodrag".

Thank you. As it turns out this question is addressed under the `nodesDraggable?` prop in the [Props List](https://reactflow.dev/docs/api/react-flow-props/#nodesdraggable) documentation.

It might be nice to add it to the [Custom Nodes API](https://reactflow.dev/docs/api/nodes/custom-nodes/#prevent-dragging--selecting) document as well, since `nodrag` and `nowheel` are documented there.

**Comment by NFTvoxies** [comment]
it's not working anymore any other solutions for the select option inside the react node ?

**Reply by bcakmakoglu** [reply]
What's not working anymore? 
`nodrag` is still a valid class name and the recommended way to suppress dragging behavior on an element inside of a custom node.
